controllers/datadogagent/feature/admissioncontroller/feature_test.go

```diff
@@ -36,70 +35,45 @@ func TestAdmissionControllerFeature(t *testing.T) {
-		},
-		{
-			Name:          "v1alpha1 admission controller enabled",
-			DDAv1:         newV1Agent(true),
-			WantConfigure: true,
-			ClusterAgent:  testDCAResources("hostip"),
-		},
 
 		//////////////////////////
 		// v2Alpha1.DatadogAgent
 		//////////////////////////
 		{
 			Name:          "v2alpha1 admission controller not enabled",
-			DDAv2:         newV2Agent(false, "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
+			DDAv2:         newV2Agent(false, "", "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
 			WantConfigure: false,
 		},
 		{
 			Name:          "v2alpha1 admission controller enabled",
-			DDAv2:         newV2Agent(true, "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
+			DDAv2:         newV2Agent(true, "", "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources(""),
+			ClusterAgent:  testDCAResources("", ""),
 		},
 		{
 			Name:          "v2alpha1 admission controller enabled, apm uses uds",
-			DDAv2:         newV2Agent(true, "", apmUDS, &v2alpha1.DogstatsdFeatureConfig{}),
+			DDAv2:         newV2Agent(true, "", "", apmUDS, &v2alpha1.DogstatsdFeatureConfig{}),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("socket"),
+			ClusterAgent:  testDCAResources("socket", ""),
 		},
 		{
 			Name:          "v2alpha1 admission controller enabled, dsd uses uds",
-			DDAv2:         newV2Agent(true, "", &v2alpha1.APMFeatureConfig{}, dsdUDS),
+			DDAv2:         newV2Agent(true, "", "", &v2alpha1.APMFeatureConfig{}, dsdUDS),
+			WantConfigure: true,
+			ClusterAgent:  testDCAResources("socket", ""),
+		},
+		{
+			Name:          "v2alpha1 admission controller enabled, add custom registry",
+			DDAv2:         newV2Agent(true, "", "testRegistryName", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("socket"),
+			ClusterAgent:  testDCAResources("", "testRegistryName"),
 		},
 	}
 
 	tests.Run(t, buildAdmissionControllerFeature)
 }
 
-func newV1Agent(enabled bool) *v1alpha1.DatadogAgent {
-	return &v1alpha1.DatadogAgent{
-		Spec: v1alpha1.DatadogAgentSpec{
-			ClusterAgent: v1alpha1.DatadogAgentSpecClusterAgentSpec{
-				Config: &v1alpha1.ClusterAgentConfig{
-					AdmissionController: &v1alpha1.AdmissionControllerConfig{
-						Enabled:                apiutils.NewBoolPointer(enabled),
-						MutateUnlabelled:       apiutils.NewBoolPointer(true),
-						ServiceName:            apiutils.NewStringPointer("testServiceName"),
-						AgentCommunicationMode: apiutils.NewStringPointer("hostip"),
-					},
-				},
-			},
-		},
-	}
-}
-
-func newV2Agent(enabled bool, acm string, apm *v2alpha1.APMFeatureConfig, dsd *v2alpha1.DogstatsdFeatureConfig) *v2alpha1.DatadogAgent {
+func newV2Agent(enabled bool, acm, reg string, apm *v2alpha1.APMFeatureConfig, dsd *v2alpha1.DogstatsdFeatureConfig) *v2alpha1.DatadogAgent {
```

Comment:
just one last nit, IMO the var name `reg` is not very intuitive, can we use `registry` (here and in `testDCAResources`)?

Reply:
@celenechang , Thanks for the feedback!
Yes, now the last commit includes `registry` instead of `reg`.